File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/OnDamageAfterMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import DamageInfo from "../DataDefine/DamageInfo";
import AbstractCreature from "../Unit/AbstractCreature";

export class OnDamageAfterMessage implements  IMessage{
    public static readonly Id = "OnDamageAfter";
    public constructor(public info:DamageInfo,public source:AbstractCreature,public target: AbstractCreature) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/OnDamageAfterMessage.cs
using ModTheSpire.DataDefine;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///造成伤害前
    public class OnDamageAfterMessage : IEventHandle
    {
        public DamageInfo DamageInfo;
        public AbstractCreature Source;
        public AbstractCreature Target;

        public OnDamageAfterMessage(DamageInfo damageInfo, AbstractCreature source, AbstractCreature target)
        {
            DamageInfo = damageInfo;
            Source = source;
            Target = target;
        }
    }
}